### DISCUSSION TITLE
Self hosted runner connection is getting closed while running GitHub job in it

### DISCUSSION THREAD

**Discussion by SameepPanigrahi** [community/community#84920] [Actions]
### Select Topic Area

Bug

### Body

I am using a self-hosted runner which is having linux OS. I have integrated the runner to run the GitHub workflow means when I will trigger the yml file of the GitHub repository it will run on the remote self-hosted runner. So, whenever I am starting the run its working fine but after some time (around 40 to 60 min) the job is getting failed with the below error.
The self-hosted runner: internalRunnersTest lost communication with the server. Verify the machine is running and has a healthy network connection

**Comment by Uerum18** [comment]
I have the same issue as well.
Using self-hosted AWS EC2 instance (on Linux) which is started and terminated by GitHub default runner. Unfortunately, it loses connection while waiting for the main job execution.
I tried different instance configurations w/o any effect. Seems like GitHub listener issue, not the self-hosted environment one.
With all this amount of similar issues reported, I hope that the fix or at least a workaround will not take too long. (Didn't find any for a while).
[Sharing the .yml file. ](https://github.com/Uerum18/python_practice_projects/blob/0409318354ae4cd75ca19110fa9fbdf794c3b763/.github/workflows/ec2-run-test.yml)

**Comment by gilM2030** [comment]
Did any of you figure out a way to solve this or what was causing the problem?
